toolkit/utils/config.py

```diff
@@ -177,33 +177,37 @@ def load_device_config(devDescription, devRevision):
     APP_MRAM_SIZE = app_size
 
 
-def load_global_config():
+def load_global_config(part_num=None, part_rev=None, jtag_iface=None, mram_iface=None):
     global DEVICE_PART_NUMBER
     global DEVICE_REVISION
     global MRAM_BURN_INTERFACE
     global JTAG_ADAPTER
     global HASHES_DB
 
-    cfg = read_global_config(CONFIG_FILE)
+    if part_num:
+        DEVICE_PART_NUMBER = getPartDescription(part_num)
+        DEVICE_REVISION = part_rev
+        MRAM_BURN_INTERFACE = mram_iface
+        JTAG_ADAPTER = jtag_iface
+    else:
+        # load from config file.
+        cfg = read_global_config(CONFIG_FILE)
 
-    # validate configuration parameters
-    # check parameter is configured...
-    checkAttribute(cfg["DEVICE"], "Revision")
+        # validate configuration parameters
+        # check parameter is configured...
+        checkAttribute(cfg["DEVICE"], "Revision")
 
-    # read hashes DB
-    hashes = read_global_config(HASHES_DB_FILE)
+        DEVICE_PART_NUMBER = cfg["DEVICE"]["Part#"]
+        DEVICE_REVISION = cfg["DEVICE"]["Revision"]
 
-    DEVICE_PART_NUMBER = cfg["DEVICE"]["Part#"]
-    DEVICE_REVISION = cfg["DEVICE"]["Revision"]
+        # read hashes DB
+        HASHES_DB = read_global_config(HASHES_DB_FILE)
```

Comment:
It looks like this HASHES_DB is not being read in if the `part_num` argument is specified.

Reply:
Yes, unsure what it's for. If possible, I want the tools Not to make any changes if parts are changed.